4 Bug fixes, some minor additions, 1 major addition

coraWhite opened the pull request:
- 4 bug fixes (number format code map, number format code 42, bg/fg color flipped, Alignment Indent not being grabbed) 
- Added support for Font strikethrough, superscript/subscript (vertAlign), font size as a float
- Added support for hyperlinks with new type.
- See comments on commits for more details.

Commit:
Updated bgcolor/fgcolor bug (flipped), and changed Font Size to a float instead of int

Commit:
Fixed a major bug

Commit:
Added logic to extract worksheet/_rels relationship files for hyperlink support. A new field in the *File struct has been added that reflects this. I matched the same logic as extracting the xmlSheets and put the logic in its own file to differentiate from the other library. Matched naming conventions and made new type xmlSheetRels

Commit:
Added support for getting Strikethrough and VerticalAlignment(superscript or subscript) in Fonts. As well as fixed the bug in Alignment Indent so that it grabs it

Commit:
Updated the test value in xmlStyle_test.go to reflect the type change for Alignment Indent. Updated a comment in xmlSheetRels. Removed one of my comments in xmlStyle.go

Commit:
Revised test cases for the field types I changed for Font Size, Indent. Added some minor test cases for Font attributes I added, including vertAlign and strike through

aweiker commented on xmlSheetRels.go:
```diff
@@ -0,0 +1,60 @@
+	"github.com/pkg/errors"
+)
+
+// DFS implementation of worksheet relationships, primarily with the focus of adding support for hyperlinks, which are located in the xl/worksheets/_rels path location for each workbook.
```
There should be nothing DFS specific in this codebase.

aweiker commented on xmlSheetRels.go:
```diff
@@ -0,0 +1,60 @@
+	return sheetRels, nil
+}
+
+// Iterate through each sheet and marshall each sheet to a map?
```
What are these comments referring to?

aweiker commented on xmlStyle.go:
```diff
@@ -187,11 +188,12 @@ func (styles *xlsxStyleSheet) getStyle(styleIndex int) *Style {
 		if xf.FontId > -1 && xf.FontId < styles.Fonts.Count {
 			xfont := styles.Fonts.Font[xf.FontId]
-			style.Font.Size, _ = strconv.Atoi(xfont.Sz.Val)
+			style.Font.Size, _ = strconv.ParseFloat(xfont.Sz.Val, 64) // returns err
```
I'm not a fan of this comment. The previous error just ignored the error if there was one.

aweiker commented on xmlStyle.go:
```diff
@@ -925,7 +947,7 @@ func (xf *xlsxXf) Marshal(outputBorderMap, outputFillMap, outputFontMap map[int]
 type xlsxAlignment struct {
 	Horizontal   string `xml:"horizontal,attr"`
-	Indent       int    `xml:"indent,attr"`
+	Indent       string `xml:"indent,attr"`
```
Why was this changed to a string?

coraWhite replied:
There was a bug where no indent values were being grabbed previously. The indent attribute is a string in the raw ooxml. It is unmarshalled into this "xlsxAlignment" type first as a string, before it is converted to int in the "Alignment" type (unchanged).

aweiker commented on style.go:
```diff
@@ -128,17 +136,19 @@ func NewFill(patternType, fgColor, bgColor string) *Fill {
-}
-
-func NewFont(size int, name string) *Font {
+	Size          float64
```
Should this be a float32 or float64?

coraWhite replied:
I attempted to use the type as float32, and it is not as succinct as float64. When converting to string from float32 and vice versa, there is an extra step of conversion since the return value of `strconv.ParseFloat()` is float64 anyway.

Commit:
Amended some minor mistakes: Removed uneccessary comments, changed Font Size from float64 to float32

Commit:
Change font size back to float64

Commit:
Add Hyperlink test file for test

Commit:
Remove unnecessary fields in xmlRel types

Commit:
Write tests for hyperlinks using test file

ghost commented:
why are you using codecov.io and not sonarQube? we are not paying for codecov.io and you shouldn't be considering your account as a personal account.

Commit:
Add Hyperlink and xlsxHyperlink types

Commit:
Make the sheetRelationships map on file un-exported

Commit:
Add Hyperlink field to each sheet if it has any

- Values of hyperlink targets now stored in each sheet
- Each sheet has a Hyperlinks attribute as a map, with the key being the cell coordinate that contains the hyperlink
- The value of the Hyperlinks map is type Hyperlink, containing metadata around the hyperlink in addition to its URL (target)

Commit:
Change the xmlSheetRel to be un-exported
